analyzers/src/SonarAnalyzer.Common/Extensions/CompilationExtensions.cs

```diff
@@ -57,4 +57,7 @@ public static bool IsMemberAvailable<TMemberType>(this Compilation compilation,
             ? memberSymbols.Any()
             : memberSymbols.Any(memberCheck);
     }
+
+    public static bool IsValidLocation(this Compilation compilation, Location location) =>
+        location.Kind != LocationKind.SourceFile || compilation.ContainsSyntaxTree(location.SourceTree);
```

Comment:
Optional: This could also be an extension method on `Location` instead. `location.IsValid(compilation)` reads a little bit better, IMO.